homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,225 @@
+"""DataUpdateCoordinator for the LG ThinQ device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from thinqconnect import (
+    ConnectBaseDevice,
+    DeviceType,
+    ThinQApi,
+    ThinQAPIErrorCodes,
+    ThinQAPIException,
+)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import COMPANY, DEVICE_TYPE_API_MAP, DOMAIN, NONE_KEY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """LG Device's Data Update Coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        thinq_api: ThinQApi,
+        device_api: ConnectBaseDevice,
+        *,
+        sub_id: str | None = None,
+    ) -> None:
+        """Initialize data coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}_{device_api.device_id}",
+        )
+        self._hass = hass
+        self._thinq_api = thinq_api
+        self._is_connected = True
+
+        # If sub_id is NONE_KEY("_") then it should be None.
+        self._sub_id = None if sub_id == NONE_KEY else sub_id
+
+        # The device name is usually set to 'alias'.
+        # But, if the sub_id exists, it will be set to 'alias {sub_id}'.
+        # e.g. alias='MyWashTower', sub_id='dryer' then 'MyWashTower dryer'.
+        self._device_name = (
+            f"{device_api.alias} {self._sub_id}" if self._sub_id else device_api.alias
+        )
+
+        # The unique id is usually set to 'device_id'.
+        # But, if the sub_id exists, it will be set to 'device_id_{sub_id}'.
+        # e.g. device_id='TQSXXXX', sub_id='dryer' then 'TQSXXXX_dryer'.
+        self._unique_id = (
+            f"{device_api.device_id}_{self._sub_id}"
+            if self._sub_id
+            else device_api.device_id
+        )
+
+        # Get the api instance.
+        self._device_api = device_api.get_sub_device(self._sub_id) or device_api
+
+    @property
+    def thinq_api(self) -> ThinQApi:
+        """Returns the thinq api."""
+        return self._thinq_api
```

Comment:
Might as well just store it in `self.thinq_api` from the beginning

Reply:
Removed